### PR TITLE
ci: conformance-[gateway-api|ginkgo|ingress] wait for images before matrix generation

### PR DESCRIPTION
**Pull request by aanm**
Followup from https://github.com/cilium/cilium/pull/34707

### CODE REVIEW

**Review by sayboras — APPROVED** [review]

File: .github/workflows/conformance-gateway-api.yaml [hunk 1]
```diff
@@ -81,8 +81,24 @@ jobs:
...
+    steps:
+      - name: Checkout context ref (trusted)
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          ref: ${{ inputs.context-ref || github.sha }}
+          persist-credentials: false
+      - name: Wait for images
+        uses: ./.github/actions/wait-for-images
+        with:
+          SHA: ${{ inputs.SHA }}
```
Comment (tklauser) [inline .github/workflows/conformance-gateway-api.yaml:98]: Shouldn't we add

```suggestion
          SHA: ${{ inputs.SHA }}
          images: cilium-ci operator-generic-ci
```

so the workflow is only waiting for the required images?

Comment (aanm) [reply .github/workflows/conformance-gateway-api.yaml:98]: oh, nice catch. Yes

**Review by tklauser — APPROVED** [review]

**Pull request merged**